keps/sig-network/4631-loadbalancerstatus/README.md

```diff
@@ -0,0 +1,1066 @@
+
+The cloud provider should set a `LoadBalancerProvisioning` condition
+immediately upon first becoming aware of a `Type: LoadBalancer`
+Service. At all times, the condition should be `True` if the cloud
+provider is waiting for the completion of an operation to bring the
+load balancer's behavior in line with the current state of the Service
+object, and `False` if it not. (There are no defined semantics for an
+`Unknown` value of this condition.)
+
+Any time any load-balancer-related field (or provider-specific
+annotation) of the Service object changes, the cloud provider must
+update at least the `LastTransitionTime` of the
+`LoadBalancerProvisioning` condition, to indicate that it has seen the
+update, even if the value of `LoadBalancerProvisioning` remains
+`False`.
+
+The well-known `Reason` values for the `LoadBalancerProvisioning`
+condition are:
+
+  - `Status=True, Reason=Provisioning`
+
+  - `Status=False, Reason=Complete`: The cloud provider has finished
+    trying to provision the load balancer (whether successfully or
+    unsuccessfully, as indicated by the `Message` and by the
+    `LoadBalancerServing` condition).
+
+#### Service Lifecycle and Interaction of the Provisioning/Serving Conditions
+
+When a load balancer Service is first created (or a non-load-balancer
+Service is converted to `Type: LoadBalancer`), one of three things
+should immediately happen:
+
+  1. The cloud provider sets `LoadBalancerProvisioning=False`,
+     `LoadBalancerServing=False`, with appropriate `Reason` values,
+     indicating that it cannot or will not provision a load balancer
+     for the service.
+
+  2. The cloud provider sets `LoadBalancerProvisioning=False`,
+     `LoadBalancerServing=True`, (and fills in `.Status.LoadBalancer`)
+     indicating that the load balancer for the service is already
+     available.
+
+  3. The cloud provider sets `LoadBalancerProvisioning=True`,
+     `LoadBalancerServing=False`, with appropriate `Reason` values,
+     indicating that it is provisioning the load balancer.
+
+In the third case, this would later be followed up by the cloud
+provider setting `LoadBalancerProvisioning` to `False` and
+`LoadBalancerServing` to either `True` or `False`, depending on
+whether the provisioning succeeded or failed.
+
+If the `LoadBalancerServing` and `LoadBalancerProvisioning` conditions
+do not appear on the Service within a reasonable amount of time after
+the Service is created (or modified to be `Type: LoadBalancer`), the
+user should assume that no cloud provider is running, and that
+`LoadBalancer` Services are not supported.
+
+(Of course, in the short term, users can't actually assume that,
+because not all cloud providers will support the new conditions.
+However, since supporting these conditions will be a prereq for
+running the `[Feature:LoadBalancer]` e2e tests, we assume that
+eventually all cloud providers will support them.)
+
+When the user updates load-balancer-relevant fields in a Service that
```

Comment:
I assume LB impls should also remove these if the service is updated in a way that removes the LB?